printing/proof_diffs.ml

```diff
@@ -251,6 +251,11 @@ let pr_letype_core goal_concl_style env sigma t =
 let pp_of_type env sigma ty =
   pr_letype_core true env sigma EConstr.(of_constr ty)
 
+let pr_leconstr_core goal_concl_style env sigma t =
+  Ppconstr.pr_lconstr_expr (Constrextern.extern_constr goal_concl_style env sigma t)
+
+let pr_lconstr_env env sigma c = pr_leconstr_core false env sigma (EConstr.of_constr c)
+
```

Comment:
This is already in `printer.ml`, right?

Reply:
Yes, it is.  Proof_diffs can't access functions in Printer because of the linking order--but I can have Printer use the copy in Proof_diffs so there aren't 2 copies.

The relationship between the two files is a bit tricky because (conceptually) the diff code prints the old and new proofs then computes the differences between the Pp.t's.  And we chose to make printing a proof with diffs part of Printer.ml (pr_open_subgoals_diff).  That seemed to make the most sense.

Reply:
OK. Maybe this means that `printer.ml` should be split into a low part and a more higher part for goals. Maybe @ejgallego has some insight of what to do here. But this starts to go beyond the scope of this PR. So, unless this new dependency is a problem for someone, I shall merge (new target) tomorrow.

Reply:
Yeah, I think that printer indeed seems to have too much stuff. IMHO a splitting into more concrete printers would help, but I decided to defer it for later due to the hassles merging the goal diff pr.